include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -1506,12 +1506,15 @@ __pattern_partial_sort_copy(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&&
         // runtime makes a dependency graph. In that case the call of __pattern_walk2 could be changed to 
         // be asynchronous for better performance.
 
+        using _ValueType = typename std::iterator_traits<_OutIterator>::value_type;
```

Comment:
It seems the first definition of `using _ValueType ` is not used (https://github.com/oneapi-src/oneDPL/pull/1599/files#diff-de507d174f2c00af6dfa912535ca543d73c6e76bf62ecf8daebfcb9293d2c904R1485)... lets remove it.

Reply:
`_ValueType` is used on the next line (below)